packages/netsuite-adapter/src/config.ts

```
@@ -179,6 +219,73 @@ const clientConfigType = createMatchingObjectType<SdfClientConfig>({
   },
 })
 
+const validateInstalledSuiteApps = (installedSuiteApps: unknown): void => {
+  validateArrayOfStrings(installedSuiteApps, [CONFIG.client, CLIENT_CONFIG.installedSuiteApps])
+  const invalidValues = installedSuiteApps.filter(id => !SUITEAPP_ID_FORMAT_REGEX.test(id))
+  if (invalidValues.length !== 0) {
+    throw new Error(`${CLIENT_CONFIG.installedSuiteApps} values should contain only lowercase characters or numbers and exactly two dots (such as com.saltoio.salto). The following values are invalid: ${invalidValues.join(', ')}`)
+  }
+}
+
+const isCustomRecordTypeName = (name: string): boolean => name.startsWith(CUSTOM_RECORD_TYPE_NAME_PREFIX)
+
+function validateMaxInstancesPerType(maxInstancesPerType: unknown):
+  asserts maxInstancesPerType is MaxInstancesPerType[] {
+  if (Array.isArray(maxInstancesPerType) && maxInstancesPerType.every(
+    val => 'name' in val && 'limit' in val && typeof val.name === 'string' && typeof val.limit === 'number'
+  )) {
+    const invalidTypes = maxInstancesPerType.filter(maxType =>
+      regex.isValidRegex(maxType.name) && noSupportedTypeMatch(maxType.name) && !isCustomRecordTypeName(maxType.name))
```

In case that the regex is invalid you still want to throw an error.